### PR TITLE
[New Rule] Azure Kubernetes Role/ClusterRole Modified or Deleted

### LINKED ISSUE #1278: [New Rule] Azure Kubernetes Role or ClusterRole Created or Modified
## Description

Identifies when a Azure Kubernetes Role/ClusterRole is Created or Modified

### Target indexes

`filebeat-*, logs-azure*`

### Platforms

Azure

### Query

```
event.dataset:azure.activitylogs and azure.activitylogs.operation_name:
	(MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/WRITE or
	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/DELETE or 
	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/BIND/ACTION or 
         MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/ESCALATE/ACTION or
	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/WRITE or 
	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/DELETE or 
	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/BIND/ACTION or
         MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/ESCALATE/ACTION) and 
event.outcome:(Success or success)
```

### **False Positives**
- 
- 

### MITRE

| Tactic | Technique ID | Technique Name | Sub-Technique Name |
| ------ | ------------ | -------------- | ------------------ |
|        |              |                |                    |
|        |              |                |                    |

### References
-

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1278

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[metadata]
+creation_date = "2021/10/18"
+maturity = "production"
+updated_date = "2021/10/18"
+integration = "azure"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when a Azure Kubernetes Role/ClusterRole is Modifiedn or Deleted.
```
Comment (w0rk3r) [inline rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml:10]: Typo, + needs elaboration, why this should be monitored?

File: rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    Kubernetes Role being modified or deleted may be performed by a system administrator. Verify whether the user identity, user agent, 
+    and/or hostname should be making changes in your environment. Kubernetes Role modified or deleted from unfamiliar users should be 
+    investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*", "logs-azure*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Azure Kubernetes Role/ClusterRole Modified or Deleted"
```
Comment (w0rk3r) [inline rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml:23]: What is the difference? There is any term that cover them both?

File: rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+query = '''
+event.dataset:azure.activitylogs and azure.activitylogs.operation_name:
+	(MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/WRITE or
+	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/DELETE or 
+	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/BIND/ACTION or 
+         MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/ESCALATE/ACTION or
+	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/WRITE or 
+	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/DELETE or 
+	 MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/BIND/ACTION or
+         MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/ROLES/ESCALATE/ACTION) and
```
Comment (w0rk3r) [inline rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml:48]: Quotes

File: rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+rule_id = "67178b42-21a6-474b-90ed-a2a8b91b2e83"
+severity = "low"
+tags = ["Elastic", "Cloud", "Azure", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+
+query = '''
+event.dataset:azure.activitylogs and azure.activitylogs.operation_name:
+	("MICROSOFT.KUBERNETES/CONNECTEDCLUSTERS/RBAC.AUTHORIZATION.K8S.IO/CLUSTERROLES/WRITE" or
```
Comment (w0rk3r) [inline rules/integrations/azure/privilege_escalation_azure_kubernetes_role_modified_or_deleted.toml:42]: Same requests as https://github.com/elastic/detection-rules/pull/1576#discussion_r732591915

**Review by w0rk3r — CHANGES_REQUESTED** [review]

**Comment by austinsonger** [comment]
I will be going on leave for the next 3 weeks. I'm going to close this and reopen when I have time.

**Pull request closed without merging**